### PR TITLE
Fix: When searching on People page, number of matches is hidden.

### PR DESCRIPTION
**Pull request by KaterinaWire**
### Issues
When searching on People page, number of matches is not hidden ("Group Members (1)" or "Group Admins (1)").

### Solutions
Add a new `isSearching` parameter  to the viewModel. If it is true, we hide the number of matches, otherwise we show it.

### CODE REVIEW

File: Wire-iOS/Sources/UserInterface/GroupDetails/GroupParticipantsDetail/GroupParticipantsDetailViewController.swift [hunk 1]
```diff
@@ -129,8 +129,8 @@ final class GroupParticipantsDetailViewController: UIViewController {
         ])
     }
     
-     func participantsDidChange() {
-        collectionViewController.sections = computeSections()
+     func participantsDidChange(isSearching: Bool) {
```
Comment (billypchan) [inline Wire-iOS/Sources/UserInterface/GroupDetails/GroupParticipantsDetail/GroupParticipantsDetailViewController.swift]: `isSearching` is not clear for the parameter's usage, I would suggest `showSectionCount`

File: Wire-iOS/Sources/UserInterface/GroupDetails/GroupParticipantsDetail/GroupParticipantsDetailViewModel.swift [hunk 2]
```diff
@@ -31,7 +31,7 @@ final class GroupParticipantsDetailViewModel: NSObject, SearchHeaderViewControll
     
     let selectedParticipants: [UserType]
     let conversation: ZMConversation
-    var participantsDidChange: (() -> Void)? = nil
+    var participantsDidChange: ((_:Bool) -> Void)? = nil
```
Comment (billypchan) [inline Wire-iOS/Sources/UserInterface/GroupDetails/GroupParticipantsDetail/GroupParticipantsDetailViewModel.swift]:
```suggestion
    var participantsDidChange: (Bool -> Void)? = nil
```

File: Wire-iOS/Resources/Base.lproj/Localizable.strings [hunk 3]
```diff
@@ -885,6 +885,8 @@
 
 "group_details.conversation_admins_header.title" = "Group admins (%d)";
 "group_details.conversation_members_header.title" = "Group members (%d)";
+"group_details.conversation_admins_header_without_parameter.title" = "Group admins";
+"group_details.conversation_members_header_without_parameter.title" = "Group members";
```
[the remark this answers is not in the document]
Comment (KaterinaWire) [reply Wire-iOS/Resources/Base.lproj/Localizable.strings]: Fixed

File: Wire-iOS/Sources/UserInterface/GroupDetails/Sections/ParticipantsSectionController.swift [hunk 4]
```diff
@@ -94,14 +93,17 @@ private struct ParticipantsSectionViewModel {
...
     ///   - maxDisplayedParticipants: max number of participants we can display, if there are more than maxParticipants participants
+    ///   - isSearching: current view model - a search result or not
     init(participants: [UserType],
          conversationRole: ConversationRole,
          totalParticipantsCount: Int,
          clipSection: Bool = true,
          maxParticipants: Int = 7,
-         maxDisplayedParticipants: Int = 5) {
+         maxDisplayedParticipants: Int = 5,
+         isSearching: Bool = false) {
```
[the remark this answers is not in the document]
Comment (KaterinaWire) [reply Wire-iOS/Sources/UserInterface/GroupDetails/Sections/ParticipantsSectionController.swift]: I will change it in another PR. These items are not only in this VC.

File: Wire-iOS/Sources/UserInterface/GroupDetails/Sections/ParticipantsSectionController.swift [hunk 5]
```diff
@@ -56,15 +56,14 @@ private struct ParticipantsSectionViewModel {
     let participants: [UserType]    
     let conversationRole: ConversationRole
     
+    var isSearching: Bool = false
```
[the remark this answers is not in the document]
Comment (KaterinaWire) [reply Wire-iOS/Sources/UserInterface/GroupDetails/Sections/ParticipantsSectionController.swift]: Fixed

File: Wire-iOS/Sources/UserInterface/GroupDetails/GroupParticipantsDetail/GroupParticipantsDetailViewModel.swift [hunk 6]
```diff
@@ -45,7 +45,11 @@ final class GroupParticipantsDetailViewModel: NSObject, SearchHeaderViewControll
     var participants = [UserType]() {
         didSet {
             self.computeParticipantGroups()
-            participantsDidChange?()
+            if let query = filterQuery {
+                participantsDidChange?(!query.isEmpty)
+            } else {
+                participantsDidChange?(false)
+            }
```
[the remark this answers is not in the document]
Comment (KaterinaWire) [reply Wire-iOS/Sources/UserInterface/GroupDetails/GroupParticipantsDetail/GroupParticipantsDetailViewModel.swift]: Fixed

**Pull request merged**